File: Proj4a/insertionSort/insertionSort.java


public class insertionSort {
   

    public static void insertionSort(int[] array) {
        for (int i = 1; i < array.length; i++) {
            int key = array[i];
            int j = i - 1;

            while (j >= 0 && array[j] > key) {
                array[j + 1] = array[j];
                j = j - 1;
            }
            array[j + 1] = key;
        }
    }

    public static void imprimeVetor(int[] array) {
        System.out.print("\n{" + array[0]);
        for (int i = 1; i < array.length; i++) {
            System.out.print(" ," + array[i]);
        }
        System.out.println("}");
    }

    public static void main(String[] args) {
        int vetor[] = {34, 2, 1, 56, 43, 8};
        insertionSort(vetor);
        imprimeVetor(vetor);
    }
}

File: Proj4a/insertionSort/insertionSort.c
#include <stdlib.h>
#include <stdio.h>


void insertionSort(int* vetor, long int tam){
    int chave;
    int j, i;
    for (i = 1; i < tam; i++){
        chave = vetor[i];
        j = i - 1;

        while (j >= 0 && vetor[j] > chave){
            vetor[j + 1] = vetor[j];
            j--;
        }
        vetor[j + 1] = chave;
    }
}

void imprimeVetor(int *vetor, long int tam){
    printf("\n{%d", vetor[0]);
    for(int i = 1; i < tam; i++){
        printf(" ,%d", vetor[i]);
    }
    printf("}");
}

int main(void){
    int vetor[] = {34, 2, 1, 56, 43, 8};
    insertionSort(vetor, 6);
    imprimeVetor(vetor, 6);
    return 0;
}